lib/living_dead.rb

```diff
@@ -2,17 +2,30 @@
 require "living_dead/living_dead"
 
 require 'stringio'
+require 'objspace'
 
 module LivingDead
 
   @string_io = StringIO.new
+  @tracing_hash = {}
+  @freed_hash   = {}
+
+  def self.tracing_hash
+    @tracing_hash
+  end
+
+  def self.freed_hash
+    @freed_hash
+  end
 
   def self.trace(*args)
-    self.start
+    # self.start
     trace = ObjectTrace.new(*args)
 
     self.tracing_hash[trace.key] = trace
     self.freed_hash[trace.key]   = false
+
+    ObjectSpace.define_finalizer(args.first, -> (object_id) { LivingDead.freed_hash[object_id] = true })
```

Comment:
Good call. Related: would like to make a tool to make a context leak like this easier to find/debug if it's happening somewhere in a larger app. Working on a `WeakRef` solution, which is yet another thing I didn't realize existed.